here-naksha-lib-hub/src/main/java/com/here/naksha/lib/hub/storages/NHSpaceStorageWriter.java

```diff
@@ -76,46 +80,83 @@ public NHSpaceStorageWriter(
-      return admin.execute(wc);
+    } else {
+      logger.info("WriteCollections Request for {}, against Custom storage.", collectionId);
+      return executeWriteToCustomSpaces(wc, collectionId);
     }
   }
 
   private @NotNull Result executeWriteFeatures(final @NotNull WriteFeatures wf) {
     final String spaceId = wf.getCollectionId();
     logger.info("WriteFeatures Request against spaceId={}", spaceId);
     addSpaceIdToStreamInfo(spaceId);
-    if (virtualSpaces.containsKey(spaceId)) {
+    if (isDeleteSpaceRequest(wf, spaceId)) {
+      return executeDeleteSpace(wf);
+    } else if (virtualSpaces.containsKey(spaceId)) {
       // Request is to write to Naksha Admin space
-      return executeWriteFeaturesToAdminSpaces(wf);
+      return executeWriteToAdminSpaces(wf, spaceId);
     } else {
       // Request is to write to Custom space
-      return executeWriteFeaturesToCustomSpaces(wf);
+      return executeWriteToCustomSpaces(wf, spaceId);
     }
   }
 
-  private @NotNull Result executeWriteFeaturesToAdminSpaces(final @NotNull WriteFeatures<?, ?, ?> wf) {
+  private @NotNull Result executeWriteToAdminSpaces(
+      final @NotNull WriteRequest<?, ?, ?> wr, @NotNull String spaceId) {
     // Run pipeline against virtual space
-    final String spaceId = wf.getCollectionId();
     final EventPipeline pipeline = pipelineFactory.eventPipeline();
     final Result result = setupEventPipelineForAdminVirtualSpace(spaceId, pipeline);
-    if (!(result instanceof SuccessResult)) return result;
-    return pipeline.sendEvent(wf);
+    if (!(result instanceof SuccessResult)) {
+      return result;
+    }
+    return pipeline.sendEvent(wr);
   }
 
-  private @NotNull Result executeWriteFeaturesToCustomSpaces(final @NotNull WriteFeatures<?, ?, ?> wf) {
-    final String spaceId = wf.getCollectionId();
+  private @NotNull Result executeWriteToCustomSpaces(
+      final @NotNull WriteRequest<?, ?, ?> wr, @NotNull String spaceId) {
     final EventPipeline eventPipeline = pipelineFactory.eventPipeline();
     final Result result = setupEventPipelineForSpaceId(spaceId, eventPipeline);
-    if (!(result instanceof SuccessResult)) return result;
-    return eventPipeline.sendEvent(wf);
+    if (!(result instanceof SuccessResult)) {
+      return result;
+    }
+    return eventPipeline.sendEvent(wr);
+  }
+
+  private boolean isDeleteSpaceRequest(@NotNull WriteFeatures<?, ?, ?> wf, @NotNull String spaceId) {
+    return NakshaAdminCollection.SPACES.equals(spaceId)
+        && wf.features.size() == 1
+        && EWriteOp.DELETE.toString().equals(wf.features.get(0).getOp());
+  }
+
+  private @NotNull Result executeDeleteSpace(@NotNull WriteFeatures<?, ?, ?> deleteSpaceEntryReq) {
+    if (deleteSpaceEntryReq.features.get(0).getFeature() instanceof Space space) {
+      WriteXyzCollections deleteSpaceReq = new WriteXyzCollections().delete(new XyzCollection(space.getId()));
```

Comment:
let's use delete of `id, uuid=null` instead of `new XyzCollection(id)` , unless we are observing any issue with lib-psql, which should be reported as well.

Reply:
When applying suggested change we get NPE on `decodeParts` from codec - let's address it in separate bugfixing ticket
Stacktrace:
```
java.lang.NullPointerException
	at com.here.naksha.lib.core.models.storage.XyzCodec.decodeParts(XyzCodec.java:52)
	at com.here.naksha.lib.core.models.storage.XyzCodec.decodeParts(XyzCodec.java:36)
	at com.here.naksha.lib.psql.PostgresWriteOp.decode(PostgresWriteOp.java:46)
	at com.here.naksha.lib.psql.PostgresSession.executeWrite(PostgresSession.java:492)
	at com.here.naksha.lib.psql.PsqlWriteSession.execute(PsqlWriteSession.java:89)
	at com.here.naksha.lib.handlers.DefaultStorageHandler.forwardWriteRequest(DefaultStorageHandler.java:196)
	at com.here.naksha.lib.handlers.DefaultStorageHandler.forwardWriteCollections(DefaultStorageHandler.java:168)
	at com.here.naksha.lib.handlers.DefaultStorageHandler.forwardRequestToStorage(DefaultStorageHandler.java:125)
	at com.here.naksha.lib.handlers.DefaultStorageHandler.processEvent(DefaultStorageHandler.java:111)
	at com.here.naksha.lib.core.EventPipeline$Event.sendUpstream(EventPipeline.java:96)
	at com.here.naksha.lib.handlers.AuthorizationEventHandler.processEvent(AuthorizationEventHandler.java:55)
	at com.here.naksha.lib.core.EventPipeline$Event.sendUpstream(EventPipeline.java:96)
	at com.here.naksha.lib.core.IEvent.sendUpstream(IEvent.java:70)
	at com.here.naksha.lib.core.EventPipeline.sendEvent(EventPipeline.java:242)
	at com.here.naksha.lib.hub.storages.NHSpaceStorageWriter.executeWriteToCustomSpaces(NHSpaceStorageWriter.java:132)
	at com.here.naksha.lib.hub.storages.NHSpaceStorageWriter.executeWriteCollections(NHSpaceStorageWriter.java:95)
	at com.here.naksha.lib.hub.storages.NHSpaceStorageWriter.executeDeleteSpace(NHSpaceStorageWriter.java:144)
	at com.here.naksha.lib.hub.storages.NHSpaceStorageWriter.executeWriteFeatures(NHSpaceStorageWriter.java:104)
	at com.here.naksha.lib.hub.storages.NHSpaceStorageWriter.execute(NHSpaceStorageWriter.java:75)
	at com.here.naksha.app.service.http.tasks.AbstractApiTask.executeWriteRequestFromSpaceStorage(AbstractApiTask.java:258)
	at com.here.naksha.app.service.http.tasks.SpaceApiTask.executeDeleteSpace(SpaceApiTask.java:116)
	at com.here.naksha.app.service.http.tasks.SpaceApiTask.execute(SpaceApiTask.java:93)
	at com.here.naksha.app.service.http.tasks.SpaceApiTask.execute(SpaceApiTask.java:51)
	at com.here.naksha.lib.core.AbstractTask.init_and_execute(AbstractTask.java:436)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at com.here.naksha.lib.core.AbstractTask$NakshaWorker.run(AbstractTask.java:96)
```

Reply:
Ok, we can fix this separately, but we need to report this for lib-psql support. Have you already merged from NAksha_maintenance? (bcos I remember Pawel already fixed this or similar issue)

Reply:
`WriteXyzCollection` can stay as object-based argument, deleting `Space` however (in Api) should rely only on id (as discussed)